src/app/clusters/ota-requestor/DefaultOTARequestor.h

```diff
@@ -111,6 +111,41 @@ class DefaultOTARequestor : public OTARequestorInterface, public BDXDownloader::
      */
     CHIP_ERROR Init(Server & server, OTARequestorStorage & storage, OTARequestorDriver & driver, BDXDownloader & downloader);
 
+    /**
+     * Called to override the default device config vendor ID value
+     */
+    void SetQueryImageVendorId(VendorId vendorId)
+    {
+        // Vendor ID 0x0000 is a reserved value
+        if (vendorId != 0)
+        {
+            mVendorId.SetValue(vendorId);
+        }
+    }
+
+    /**
+     * Called to override the default device config product ID value
+     */
+    void SetQueryImageProductId(uint16_t productId)
+    {
+        // Product ID 0x0000 is a reserved value
+        if (productId != 0)
+        {
+            mProductId.SetValue(productId);
+        }
+    }
+
+    /**
+     * Called to override the default device config hardware version value
+     */
+    void SetQueryImageHardwareVersion(uint16_t hwVersion)
```

Comment:
The OTARequestorInterface  should have getter interface methods for these (and for the provider metadata). Then, DefaultOTARequestor concrete class could have the setters just in the concrete class (i.e. just like they are now with no changes), but `TriggerImmediateQueryInternal` could use the getters to get the values. This would allow overriding the method by which these are set.